components/event-service/internal/events/subscribed/events.go

```diff
@@ -34,84 +33,63 @@ type Event struct {
 }
 
 type eventsClient struct {
-	subscriptionsClient SubscriptionsGetter
-	namespacesClient    NamespacesClient
+	triggerLister kneventinglister.TriggerLister
 }
 
 //NewEventsClient function creates client for retrieving all active events
-func NewEventsClient(subscriptionsClient SubscriptionsGetter, namespacesClient NamespacesClient) EventsClient {
+func NewEventsClient(client versioned.Interface) EventsClient {
+	informerFactory := externalversions.NewSharedInformerFactory(client, 0)
+	ctx := signals.NewContext()
+	go informerFactory.Start(ctx.Done())
+	informerFactory.WaitForCacheSync(ctx.Done())
 
 	return &eventsClient{
-		subscriptionsClient: subscriptionsClient,
-		namespacesClient:    namespacesClient,
+		triggerLister: informerFactory.Eventing().V1alpha1().Triggers().Lister(),
 	}
 }
 
 func (ec *eventsClient) GetSubscribedEvents(appName string) (Events, error) {
-	var activeEvents []Event
-
-	namespaces, e := ec.getAllNamespaces()
-
-	if e != nil {
-		return Events{}, e
-	}
-
-	for _, namespace := range namespaces {
-		events, err := ec.getEventsForNamespace(appName, namespace)
-		if err != nil {
-			return Events{}, err
-		}
-		activeEvents = append(activeEvents, events...)
+	activeEvents, err := ec.getKnativeTriggers(appName)
+	if err != nil {
+		return Events{}, err
 	}
-
 	activeEvents = removeDuplicates(activeEvents)
-
 	return Events{EventsInfo: activeEvents}, nil
 }
 
-func (ec *eventsClient) getEventsForNamespace(appName, namespace string) ([]Event, error) {
-	subscriptionList, e := ec.subscriptionsClient.Subscriptions(namespace).List(meta.ListOptions{})
-
-	if e != nil {
-		return nil, e
+func (ec *eventsClient) getKnativeTriggers(appName string) ([]Event, error) {
```

Comment:
`getKnativeTriggers()` returns events, not triggers. I'd consider a new method name.

Reply:
refactored to *getEventList*